src/Workspaces/Core/Portable/Utilities/Operations/IOperationContext.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Utilities
```

Comment:
>Utilities [](start = 33, length = 9)

Utilities is not a great public namespace. Do we already have any public API in this subnamespace? If not let's just put the type to Microsoft.CodeAnalysis.

Reply:
that's a good question.  i'll see if we can put it elsewhere (or in the root). I'm a little concerned though with the potential naming confusion/clash with IOperation.  Thoughts @333fred ?

Reply:
Yes, but such naming collision is not really addressed by adding a namespace. I think we should rename this type to be more specific. "Operation" and "Context" are both very generic names. They don't tell me at all what this type is actually good for.

---
In reply to: [551563491](https://github.com/dotnet/roslyn/pull/50211#discussion_r551563491) [](ancestors = 551563491)

Reply:
Something like `LongRunningOperationProgress`

---
In reply to: [551565822](https://github.com/dotnet/roslyn/pull/50211#discussion_r551565822) [](ancestors = 551565822,551563491)

Reply:
i like that :)

Reply:
> I'm a little concerned though with the potential naming confusion/clash with IOperation.

Agreed. We should not prefix these type names with `IOperation` as they are completely unrelated to `IOperation` API in the compiler layer and can cause confusion.

Reply:
I also Tomas's proposed name, in the CodeAnalysis namespace directly.